vtex-sites/nextjs.store
# feat: Bun compatibility

## What's the purpose of this pull request?
This PR makes this starter compatible with the new JS runtime called [bun](https://bun.sh/)

Bun is the new JS runtime compatible with both Node.JS and Web APIs. It promises to drastically improve performance for both runtime and bundling/installing dependencies. Below, I've gathered some metrics on the differences between our current stack (node + yarn) vs bun. These are values for 95% confidence interval

|      runtime      |     install     | node_modules |  build (fresh) |  build (cache)  |
|:-----------------:|:---------------:|:------------:|:--------------:|:---------------:|
| yarn1.22 + node18 | 40.5186s ±1.711 |     800M     | 59.7888s ±3.89 | 18.0767s ±2.471 |
|      bun0.1.1     |  4.3079s ±0.551 |     709M     |  36.7s ±2.618  | 11.9764s ±0.535 |

## How does it work?
`bun` is not yet a drop in replacement for node. You can notice by the dependencies I've had to remove to make this starter compatible with `bun`. Hopefully, these dependencies were not need by either Next.JS or Storybook and I think it's safe to merge this PR and remove these dependencies anyways. 

I've had issues with the gatsby.store starter, where the sharp module would not be installed. Maybe this will be fixed in the future and we will be able to use bun in there as well.

Vercel is preparing to provide the `bun` runtime and I guess we should prepare ourselves too by adding early support on FastStore Cloud platform via a flag. I think we would have massive gains on this regard.

## How to test it?
Download this starter and run `bun install && bun run develop`. You should see the starter as usual.

## References
- https://bun.sh/
